components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/remote/httpclient/HttpClientManagerImpl.java

```diff
@@ -88,6 +108,23 @@ public void release(final ProxyRepository proxyRepository, final RemoteStorageCo
   // ==
 
   /**
+   * Method that returns {@code true} if System property with key {@code propertyKey} contains comma delimited string
+   * list, and the passed in {@code proxyRepository} remote URL' hostname is enlisted in it.
+   */
+  protected boolean isProxiedHostnameEnlisted(final ProxyRepository proxyRepository, final String propertyKey) {
+    try {
+      final URI uri = new URI(proxyRepository.getRemoteUrl());
+      final List<String> enlistedIds = Lists.newArrayList(
+          Splitter.on(",").split(SystemPropertiesHelper.getString(propertyKey, "")));
```

Comment:
Might be worth doing this in a static initialization block rather than continually parse and rebuild the same list every time we create a new HttpClient